engine/src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -223,6 +224,13 @@ class WorkflowActor(val workflowId: WorkflowId,
     case Event(WorkflowExecutionFailedResponse(executionStore, outputStore, failures),
     data @ WorkflowActorData(_, Some(workflowDescriptor), _, _)) =>
       finalizeWorkflow(data, workflowDescriptor, executionStore, outputStore, Option(failures.toList))
+    case Event(msg @ EngineStatsActor.JobCountQuery, data) =>
```

Comment:
ToL If you put this in `whenUnhandled` you could respond even if we're no longer in `executing` (and your "should be impossible" will be possible!)